clientcredentialsx/auth.go

```diff
@@ -0,0 +1,82 @@
+package clientcredentialsx
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"go.infratographer.com/x/viperx"
+
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// Auth is an oauth2 client config
+type Auth struct {
+	config *clientcredentials.Config
+	ctx    context.Context
+}
+
+// MustViperFlags adds oidc oauth2 client config to the provided flagset and binds to viper
+func MustViperFlags(v *viper.Viper, flags *pflag.FlagSet) {
+	flags.Bool("oidc-client", true, "use oidc client auth")
+	viperx.MustBindFlag(v, "oidc.client.enabled", flags.Lookup("oidc-client"))
+
+	flags.String("oidc-client-id", "", "expected oidc client identifier")
+	viperx.MustBindFlag(v, "oidc.client.id", flags.Lookup("oidc-client-id"))
+
+	flags.String("oidc-client-secret", "", "expected oidc client secret")
+	viperx.MustBindFlag(v, "oidc.client.secret", flags.Lookup("oidc-client-secret"))
+
+	flags.String("oidc-client-token-url", "", "expected oidc token url")
+	viperx.MustBindFlag(v, "oidc.client.token-url", flags.Lookup("oidc-client-token-url"))
+}
+
+// configFromViper returns an oauth2 client credentials config from the provided viper config
+func configFromViper(v *viper.Viper) (*clientcredentials.Config, error) {
```

Comment:
I think per a few other conversations, we're actually trying to avoid the `configFromViper` helpers. More details below

https://github.com/infratographer/x/pull/87#discussion_r1218139535